File: nostos-core/src/main/java/oui/sncf/io/coordinates/Longitude.java
package oui.sncf.io.coordinates;

class Longitude {

    public static final int MAX_DEGREES = 180;
    public static final int MIN_DEGREES = -180;

    private final double degrees;

    private Longitude(final double degrees) {
        this.degrees = degrees;
    }

    public static Longitude of(final double degrees) {
        if (degrees <= MIN_DEGREES || degrees >= MAX_DEGREES) {
            throw new IllegalArgumentException("longitude should be between -180° and 180°");
        }

        return new Longitude(degrees);
    }

    public double getDegrees() {
        return degrees;
    }
}

File: nostos-core/src/main/kotlin/io/fries/nostos/core/coordinates/Longitude.kt
package io.fries.nostos.core.coordinates

internal class Longitude(val degrees: Double) {

    init {
        require(degrees in MIN_DEGREES..MAX_DEGREES) { "Longitude degrees should be in the range [$MIN_DEGREES, $MAX_DEGREES]" }
    }

    companion object {
        private const val MIN_DEGREES = -180.0
        private const val MAX_DEGREES = 180.0
    }
}